code/createDeviceSettingsTable.m

```diff
@@ -0,0 +1,757 @@
+streamStartCounter_FFT = 1; % Initalize counter for streaming starts
+streamStopCounter_FFT = 1; % Initalize counter for streaming stops
+
+while recordCounter <= length(DeviceSettings)
+    currentSettings = DeviceSettings{recordCounter};
+    
+    % Check if SensingConfig was updated
+    if isfield(currentSettings,'SensingConfig')
+        
+        % If timeDomain updated in this record, create
+        % new row in TD_SettingsTable and populate with metadata
+        if isfield(currentSettings.SensingConfig,'timeDomainChannels')
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            TD_SettingsTable.action{entryNumber_TD} = 'Sense Config';
+            TD_SettingsTable.recNum(entryNumber_TD) = NaN;
+            TD_SettingsTable.time{entryNumber_TD} = HostUnixTime;
+            
+            % Settings will remain in TDsettings until updated
+            TDsettings = convertTDcodes(currentSettings.SensingConfig.timeDomainChannels);
+            for iChan = 1:4
+                fieldName = sprintf('chan%d',iChan);
+                TD_SettingsTable.(fieldName){entryNumber_TD} = TDsettings(iChan).chanFullStr;
+            end
+            TD_SettingsTable.tdDataStruc{entryNumber_TD} = TDsettings;
+            
+            entryNumber_TD = entryNumber_TD + 1;
+        end
+        
+        % If power domain updated in this record, create
+        % new row in Power_SettingsTable and populate with metadata
+        if isfield(currentSettings.SensingConfig,'powerChannels')
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            Power_SettingsTable.action{entryNumber_Power} = 'Sense Config';
+            Power_SettingsTable.recNum(entryNumber_Power) = NaN;
+            Power_SettingsTable.time{entryNumber_Power} = HostUnixTime;
+            
+            % This populates information about the powerBands (coded
+            % values, not in Hz). The bipolar channels associated with the
+            % bands are defined above (from timeDomainChannels). Sense chan1: Bands
+            % 1-2, sense chan2: Bands 3-4, Sense chan3: Bands 5-6, Sense
+            % chan4: Bands 7-8
+            
+            % Settings will remain in powerChannels, TDsampleRate, and fftConfig until
+            % updated; TDsampleRate and fftConfig needed in later processing for
+            % determinig powerBands
+            powerChannels = currentSettings.SensingConfig.powerChannels;
+            Power_SettingsTable.powerBands{entryNumber_Power} = powerChannels;
+            
+            % Get sample rate for each TD channel; all TD channels have
+            % same Fs (or is listed as NaN)
+            for iChan = 1:4
+                TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+            end
+            TDsampleRates = unique(TDsampleRates);
+            currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
```

Comment:
Is it possible to have two different sample rates? Perhaps 500 for subcortical and 250 for subcortical electrodes? If so, the current table would not show which sampling rate is related to which channels. The easiest solution to this issue would be to write all the sampling rates to the cell. If this is type of setup is not possible ignore this nonsense!

Reply:
no it's not possible